src/libraries/Native/Unix/System.Native/pal_threading.c

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_threading.h"
+
+#include <limits.h>
+#include <sched.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include <pthread.h>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// LowLevelMonitor - Represents a non-recursive mutex and condition
+
+struct LowLevelMonitor
+{
+    pthread_mutex_t Mutex;
+    pthread_cond_t Condition;
+#ifdef DEBUG
+    bool IsLocked;
+#endif
+};
+
+#ifdef DEBUG
+static void SetIsLocked(LowLevelMonitor* monitor, bool isLocked)
+{
+    assert(monitor->IsLocked != isLocked);
+    monitor->IsLocked = isLocked;
+}
+#else
+#define SetIsLocked(monitor, isLocked)
```

Comment:
Is using empty defines like this in place of an empty function body a common pattern in System.Native?

Reply:
There are some existing instances of this pattern, e.g. https://github.com/dotnet/runtime/blob/master/src/libraries/Native/Unix/Common/pal_utilities.h#L36

I do not have opinion about this. Happy to change to something else.

Reply:
It's common and sensible when you have something that's a macro in both cases, like what you linked, but I personally don't like defining a function in one case and a macro in the other. It's not terribly important, so if it's an established pattern that's fine, but it can lead to unintuitive behavior when nesting macros so I thought I'd mention it.